lib/Doctrine/DBAL/Migrations/Tools/Console/Command/GenerateCommand.php

```diff
@@ -127,4 +136,30 @@ protected function generateMigration(Configuration $configuration, InputInterfac
 
         return $path;
     }
+
+    protected function manageCustomTemplate(Configuration $configuration, InputInterface $input, OutputInterface $output)
+    {
+        $customTemplate = $input->getOption('template') ? $configuration->getCustomTemplate() : null;
+
+        if ($customTemplate !== null) {
+            $this->loadTemplateFile($customTemplate, $output);
+        }
+    }
+
+    private function loadTemplateFile(string $customTemplate, OutputInterface $output): void
+    {
+        $customTemplate = getcwd() . '/' . $customTemplate;
+        if ( ! is_file($customTemplate)) {
+            throw new \InvalidArgumentException('The specified template « ' . $customTemplate . ' » cannot be found.');
+        }
+
+        $templateContent = file_get_contents($customTemplate);
+
+        if ($templateContent === false) {
+            throw new \InvalidArgumentException('Cannot read file contents of template « ' . $customTemplate . ' ».');
+        }
+
+        $output->writeln(sprintf('Using custom migration template "<info>%s</info>"', $customTemplate));
+        self::$_template = $templateContent;
```

Comment:
@stof : I agree this should be only on object instance, but it need to check every methods of current command to avoid a break.

I'll see this point when comming back from holidays (in a week)